### PR TITLE
store and write proxy config to wp-config.php

### PR DESCRIPTION
**Pull request by Thanhphan1147**
During `_reconciliation` the charm checks if the themes and plugins currently installed matches the expected themes and plugin. If there are some missing the charm will then download them from `https://wordpress.org`. Behind the scene it runs `wp theme|plugin install` and If the charm is deployed behind a proxy and the proxy information is not present in `wp-config.php` this command will fail, leaving the charm in a blocked state. 

This PR addresses that issue by fetching the proxy information from `juju model-config` and populate the `proxy-config` attribute of the charm state so that the values can be added to `wp-config.php` during reconciliation.

### Checklist

- [x] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The changes are compliant with [ISD054 - Managing Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [x] The documentation is generated using `src-docs`
- [x] The documentation for charmhub is updated.
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

### CODE REVIEW

**Comment by amandahla** [comment]
Could you fill the PR description please? Out of curiosity: whats the impact if the proxy is not reachable? Is Wordpress still up?

**Comment by Thanhphan1147** [comment]
Hi, Thanks for the notice, I have updated the PR description with relevant information. Basically wordpress will be in a blocked state if the plugins|themes cannot be installed

**Comment by jdkandersson** [comment]
Technically the checkbox for " The PR is tagged with appropriate label (urgent, trivial, complex)" is checked though the PR is not labelled with any of those tags, could that please be resolved?

File: src/state.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        """Initialize a new instance of the CharmConfigInvalidError exception.
+
+        Args:
+            msg: Explanation of the error.
+        """
+        self.msg = msg
+
+
+class ProxyConfig(BaseModel):
+    """Configuration for accessing Jenkins through proxy.
```
Comment (jdkandersson) [inline src/state.py:35]: Jenkins? 😉

File: tests/unit/conftest.py [hunk 2]
```diff
@@ -314,3 +315,11 @@ def attach_storage(
...
     yield
     patch.container.fs["/proc/mounts"] = ""
+
+
+@pytest.fixture(scope="function", name="reset_proxy_env")
+def reset_proxy_env():
+    """Reset proxy related env variables during charm init."""
+    os.environ["JUJU_CHARM_HTTP_PROXY"] = ""
+    os.environ["JUJU_CHARM_HTTPS_PROXY"] = ""
+    os.environ["JUJU_CHARM_NO_PROXY"] = ""
```
Comment (jdkandersson) [inline tests/unit/conftest.py:325]: If you use https://docs.pytest.org/en/7.1.x/how-to/monkeypatch.html then pytest will automatically reverse any changes made after the test completes

File: src/state.py (hunk 1 above)
Comment (Thanhphan1147) [reply src/state.py:35]: oops, I fixed the docstring.

File: tests/unit/conftest.py (hunk 2 above)
Comment (Thanhphan1147) [reply tests/unit/conftest.py]: Thanks, I updated the tests to use monkeypatch.setenv

**Review by amandahla — APPROVED** [review]
LGTM

**Review by jdkandersson — APPROVED** [review]

**Comment by Thanhphan1147** [comment]
> Technically the checkbox for " The PR is tagged with appropriate label (urgent, trivial, complex)" is checked though the PR is not labelled with any of those tags, could that please be resolved?

I labelled the PR appropriately,

**Pull request merged**